tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py

```diff
@@ -0,0 +1,236 @@
+#       'tags': '' }}
+# {'audios': ['./path/to/audio/2218.flac'],
+#  'text': '<audio>\n'
+#          '[[caption]]: Someone is ringing a bell. <|__dj__eoc|>',
+#  '__dj__meta__': {
+#       'num_captions_per_audio': 1,
+#       'title': 'Service Bell Help',
+#       'description': 'Customer ringing service bell in need of help in a store.',  # noqa: E501
+#       'author': 'Daniel Simion',
+#       'href': '2218-Service-Bell-Help.html',
+#       'caption': 'Someone is ringing a bell.',
+#       'id': '2218',
+#       'duration': 1.5698125,
+#       'audio': 'wav_path',
+#       'download_link': 'http://soundbible.com/grab.php?id=2218&type=wav',
+#       'category': '',
+#       'tags': '' }}
+
+import json
+import os
+from typing import List, Union
+
+import fire
+import jsonlines as jl
+from loguru import logger
+from tqdm import tqdm
+
+from data_juicer.utils.constant import Fields
+from data_juicer.utils.mm_utils import SpecialTokens
+
+
+def creat_meta_filed(num_captions_per_audio, source_meta):
+    meta_dict = {
+        'num_captions_per_audio': num_captions_per_audio,
+        'title': '',
+        'description': '',
+        'author': '',
+        'href': '',
+        'caption': '',
+        'id': '',
+        'duration': '',
+        'audio': '',
+        'download_link': '',
+        'category': '',
+        'tags': ''
+    }
+    for key in source_meta:
+        meta_dict[key] = source_meta[key]
+    return meta_dict
+
+
+def get_all_files(dirname):
+    result = {}
+    for maindir, subdir, file_name_list in os.walk(dirname):
+        for filename in file_name_list:
+            filepath = os.path.join(maindir, filename)
+            result[filename] = filepath
+    return result
+
+
+@logger.catch
+def main(
+    wavcaps_json_path: str,
+    wavcaps_audio_path: str,
+    target_ds_path: str,
+    target_field: Union[str, List[str]] = 'caption',
+    eoc_special_token: str = SpecialTokens.eoc,
+    audio_special_token: str = '<audio>',
+    add_eoc_at_last: bool = True,
+    add_target_field_token: bool = True,
+    sent_seperator: str = '\n',
+):
+    """
+    Convert a WavCaps-like dataset to the Data-Juicer format.
+
+    :param wavcaps_json_path: path to the json files of WavCaps-like dataset.
+    :param wavcaps_audio_path: path to the audio files of WavCaps-like dataset.
+    :param target_ds_path: path to store the converted dataset in Data-Juicer
+        format.
+    :param target_field: the field used to describe audio in the WavCaps-like
+        dataset, which can be one or more of ['caption','title','description'].
+    :param eoc_special_token: the special token for "end of a chunk". It's used
+        to split conversation chunks explicitly. Default: <|__dj__eoc|> (from
+        Data-Juicer).
+    :param audio_special_token: the special token for audios. It's used to
+        locate the audios in the text. In typical WavCaps-like datasets,
+        this token always be "<audio>". You can change it to align with your
```

Comment:
As the above comment says, it seems that WavCaps didn't specify any tokens for audio.